infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -279,7 +284,14 @@ type DefaultConfigs struct {
 }
 
 // OptionalConfigs k8s container settings that are optional
-type OptionalConfigs struct {
+type RegistryOptionalConfigs struct {
+	Env             *[]corev1.EnvVar             `json:"env,omitempty"`
+	ImagePullPolicy *corev1.PullPolicy           `json:"imagePullPolicy,omitempty"`
+	Resources       *corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type StoreOptionalConfigs struct {
+	Replicas        *int32                       `json:"replicas,omitempty"`
```

Comment:
since we want to set a default of 1 in the applied status, let's instead move this new field to a new `StoreDefaultConfigs` struct... that includes the existing `DefaultConfigs` inline. Then, look into leveraging the existing `ApplyDefaultsToStatus` in services package to apply said default to both stores (but only if nil in the spec).

this should be a relatively small code change. most of the other changes you've made to services functions should be reverted.